libraries/AP_HAL_Linux/VideoIn.cpp

```diff
@@ -250,6 +250,42 @@ void VideoIn::prepare_capture()
     }
 }
 
+void VideoIn::shrink_8bpp(uint8_t *buffer, uint8_t *new_buffer,
+                          uint32_t width, uint32_t height, uint32_t left,
+                          uint32_t selection_width, uint32_t top,
+                          uint32_t selection_height, uint32_t fx, uint32_t fy)
+{
+    uint32_t i, j, k, px, block_x, block_y, block_position;
+    uint32_t out_width = selection_width / fx;
+    uint32_t out_height = selection_height / fy;
+    uint32_t width_per_fy = width * fy;
+    uint32_t fx_fy = fx + fy;
+    uint32_t out_width_sum = 0;
+
+    /* selection offset */
+    block_y = left + top * width;
+
+    for (i = 0; i < out_height; i++) {
+        block_x = 0;
+        for (j = 0; j < out_width; j++) {
+            px = 0;
+
+            for (k = 0; k < fx; k++)
+                px += buffer[block_position + k];
+
+            for (k = 0; k < fy; k++)
+                px += buffer[block_position + width + k];
```

Comment:
here I assume that it should be
<code>
px += buffer[block_position + width * k];